### ISSUE TITLE
Unexpected 'expected indentation' errors.

### ISSUE THREAD

**Issue by nevdelap** [sqlfluff/sqlfluff#4792]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Unexpected 'expected indentation' errors.

### Expected Behaviour

While this passes...
```
USE ROLE ACCOUNTADMIN;
CREATE USER TEMP_TESTING;
DROP USER TEMP_TESTING;  -- It doesn't complain about this.
```

### Observed Behaviour

this is the minimal example that fails with sqlfluff, version 1.4.5.
```
USE ROLE ACCOUNTADMIN;
CREATE USER TEMP_TESTING PASSWORD = 'asdsadasd';
DROP USER TEMP_TESTING;  -- It shouldn't complain about this.
```
which produces...
```
== [sql/one-off/2023-04-17_AN-829_create_snowflake_objects_for_hex.sql] FAIL
L:   3 | P:   1 | L003 | Expected 1 indentation, found 0 [compared to line 02]
```

### How to reproduce

Is in Observed Behaviour.

### Dialect

snowflake

### Version

1.4.5

### Configuration

```
[sqlfluff]
# See https://docs.sqlfluff.com/en/stable/rules.html
exclude_rules = L032, L034, L051
large_file_skip_byte_limit = 50000

[sqlfluff:layout:type:comma]
line_position = leading
spacing_before = touch

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true

[sqlfluff:templater:dbt]
project_dir = /sql
profiles_dir = /.dbt
profile = default
target = dev

[sqlfluff:indentation]
indented_joins = True
template_blocks_indent = True

[sqlfluff:rules]
indent_unit = space
max_line_length = 120
tab_space_size = 4

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L024]

[sqlfluff:rules:L030]
extended_capitalisation_policy = upper

[sqlfluff:rules:L040]
capitalisation_policy = upper

[sqlfluff:rules:L063]
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
With the latest `sqlfluff` there is no more `LT02` / `L003` rule violation.
